### DISCUSSION TITLE
Help with function to filter Model ScalarFieldEnum keys

### DISCUSSION THREAD

**Discussion by dariusj18** [prisma/prisma#9670] [Q&A]
I am trying to create a function to take the ScalarFieldEnum object that prisma generates for a Model and return an array of the fields minus ones listed.  I have a start to something that seems to work, but I'm not sure if it's the best way to do it, 

```typescript
const filterFields = <T = Record<string, string>>(fieldsEnum: T, fields: (keyof T)[]): ((keyof T)[]) => {
  const keys = Object.keys(fieldsEnum).filter((field) => !fields.includes(field as keyof T)) as (keyof T)[];
  return keys;
};
```

Let us assume a model:

```prisma
model User {
  id String
  email String
  password String
}
```

generates

```typescript
export const UserScalarFieldEnum: {
  id: 'id',
  email: 'email',
  password: 'password'

export type UserScalarFieldEnum= (typeof UserScalarFieldEnum)[keyof typeof UserScalarFieldEnum]
```

and I want to run `filterFields(Prisma.UserScalarFieldEnum, ['password'])` and have it return `['id', 'email']`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
